services/alerta/client.go

```diff
@@ -0,0 +1,218 @@
+package alerta
+
+import (
+	"bytes"
+	"context"
+	"encoding/json"
+	"fmt"
+	"io"
+	"net/http"
+	"sort"
+	"strings"
+	"time"
+
+	"github.com/cenkalti/backoff"
+	"github.com/rudderlabs/rudder-server/config"
+	"github.com/rudderlabs/rudder-server/utils/httputil"
+)
+
+type OptFn func(c *ClientImpl)
+
+type Tags []string
+
+type Service []string
+
+type Alert struct {
+	Resource    string   `json:"resource"`    // warehouse-upload-aborted
+	Event       string   `json:"event"`       // CRITICAL/rudder/<resource>:<group>
+	Environment string   `json:"environment"` // PRODUCTION,DEVELOPMENT,PROXYMODE,CARBONCOPY
+	Severity    Severity `json:"severity"`    // warning,critical,normal // Get the full list from https://docs.alerta.io/api/alert.html#severity-table
+	Group       string   `json:"group"`       // cluster=rudder,destID=27CHciD6leAhurSyFAeN4dp14qZ,destType=RS,namespace=hosted,notificationServiceMode=CARBONCOPY,priority=P1,sendToNotificationService=true,team=warehouse
+	Text        string   `json:"text"`        // <event> is CRITICAL warehouse-upload-aborted-count: 1
+	Service     Service  `json:"service"`     // {upload_aborted}
+	Timeout     int      `json:"timeout"`     // 86400
+	Tags        Tags     `json:"tags"`        // {sendToNotificationService=true,notificationServiceMode=CARBONCOPY,team=warehouse,priority=P1,destID=27CHciD6leAhurSyFAeN4dp14qZ,destType=RS,namespace=hosted,cluster=rudder}
+}
+
+type Priority string
+
+type Severity string
+
+const (
+	SeverityCritical Severity = "critical"
+	SeverityWarning  Severity = "warning"
+	SeverityNormal   Severity = "normal"
+	SeverityOk       Severity = "ok"
+)
+
+const (
+	PriorityP1 Priority = "P1"
+	PriorityP2 Priority = "P2"
+	PriorityP3 Priority = "P3"
+)
+
+var (
+	defaultTimeout    = 30 * time.Second
+	defaultMaxRetries = 3
+	defaultSeverity   = SeverityCritical
+	defaultPriority   = PriorityP1
+	defaultTeam       = "No Team"
+)
+
+type Client interface {
+	SendAlert(ctx context.Context, resource string, service Service) error
+}
+
+type ClientImpl struct {
```

Comment:
[minor] naming suggestion:

```suggestion
type AlertSender interface {
	SendAlert(ctx context.Context, resource string, service Service) error
}

type Client struct {
```

Also consider defining the interface on the consumer.